### DISCUSSION TITLE
Is there a way to completely burn/discard an object?

### DISCUSSION THREAD

**Discussion by 0xbe1** [aptos-labs/aptos-developer-discussions#253] [Questions] [answered]
### Describe your question in detail.

I want to figure out a way to burn/discard an object so that it's not revertable.

### What have you tried or looked at?  Or how can we reproduce the error?

There's a [`burn` method](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/object.move#L616) in the object module however the object can be unburnt by calling the [`unburn` method](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/object.move#L625).

Wonder if there's a way to ensure the object is not redeemable by the original owner.

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by 0xbe1** [comment]
Seems this recent PR addresses it

https://github.com/aptos-labs/aptos-core/pull/13495

**Reply by kshitijc1506** [reply, marked as answer]
Hey @0xbe1, this PR ensures that the original owner **cannot** reclaim an object after a soft burn, but only if `TransferRef` is used before reclaim.

^This was originally possible since we didn't remove the `TombStone` struct in `object::transfer_with_ref`.

To address your original question, a user can still call `burn` and then `unburn`. 
 
If you want to make sure that the object is discarded, you should use the [`DeleteRef`](https://preview.aptos.dev/en/build/smart-contracts/aptos-standards/aptos-object#object-capabilities-refs). This will remove the object from global storage instead of transferring it to the burn address.

**Reply by 0xbe1** [reply]
@kshitjee ty for clarification. tbh it is confusing that `burn` doesn't really burn an object, wonder if we should make it clear in the document that it is soft deletion and people should use `delete` for a real deletion?

**Reply by kshitijc1506** [reply]
That's a fair assessment. Thanks for the feedback, will have a PR in for this soon.

**Reply by kshitijc1506** [reply]
@0xbe1 note that _named_ objects cannot be deleted. If using named objects is crucial for your use case, a possible workaround is to create a boolean flag and store it in the object.

I will also be adding this information to the documentation later, as I noticed that the only place it is currently mentioned (AFAIK) is in the comments above `object::create_named_object`.
